shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Val.hs

```diff
@@ -74,23 +73,22 @@ infixl 6 <->
 
 infixl 7 <×>
 
-(<+>) :: Val t => t -> t -> t
-x <+> y = x <> y
-
-(<->) :: Val t => t -> t -> t
-x <-> y = x <+> (invert y)
-
-(<×>) :: Val t => Int -> t -> t
-x <×> y = scale x y
+scale :: (Val t, Integral i) => i -> t -> t
+scale i v = i <×> v
 
 sumVal :: (Foldable t, Val v) => t v -> v
 sumVal xs = foldl (<+>) mempty xs
 
+invert :: Val t => t -> t
```

Comment:
Can we not just re-export this from `Data.Group`? Or you want it specialised?

Reply:
I want is specialised. Better error messages.